vsphere/internal/virtualdevice/virtual_machine_network_interface_subresource.go

```diff
@@ -870,7 +870,7 @@ func (r *NetworkInterfaceSubresource) assignEthernetCard(l object.VirtualDeviceL
 
 	// Now that we know which units are used, we can pick one
 	newUnit := int32(r.Index) + pciDeviceOffset
-	if units[newUnit] {
+	if units[newUnit-pciDeviceOffset] {
```

Comment:
Do we need to add `pciDeviceOffset` if we are just subtracting it 2 lines later? Line 872 for reference

Reply:
Yeah, `units` is a `[]bool` of length 10 that we use to get a "map" of what PCI devices in the available network device PCI range (7-16) are currently taken to make sure we are not going to add/assign a device in an already occupied slot. This obviously has an index range of 0-9, and is not reflective of the final unit number. This was actually an error as the slice is being correctly populating this list [here](https://github.com/terraform-providers/terraform-provider-vsphere/blob/fc385d285d9c8438c713a85ec80cac14b694673b/vsphere/internal/virtualdevice/virtual_machine_network_interface_subresource.go#L868), but not being subsequently referenced correctly.

This is why only 3 devices can be added right now as per the referenced issue:

```
network_interface.0 7
network_interface.1 8
network_interface.2 9
network_interface.3 10 (out of range here)
```

Reply:
Ah. Yep! Makes sense now. Thank you